src/NServiceBus.Core/Recoverability/Recoverability.cs

```diff
@@ -28,66 +29,65 @@ public Recoverability()
         protected internal override void Setup(FeatureConfigurationContext context)
         {
             var errorQueue = context.Settings.ErrorQueueAddress();
             context.Settings.Get<QueueBindings>().BindSending(errorQueue);
 
-            var failureInfoStorage = new FailureInfoStorage(context.Settings.Get<int>(FailureInfoStorageCacheSizeKey));
-            var localAddress = context.Settings.LocalAddress();
+            var transportTransactionMode = context.Settings.GetRequiredTransactionModeForReceives();
+
+            if (transportTransactionMode == TransportTransactionMode.None)
+            {
+                Logger.Warn("First and Second Level Retries will be disabled. Automatic retries are not supported when running with TransportTransactionMode.None. Failed messages will be moved to the error queue instead.");
+            }
 
-            context.Pipeline.Register("Recoverability", b =>
+            context.Container.ConfigureComponent(b =>
             {
-                var hostInfo = b.Build<HostInformation>();
-                var staticFaultMetadata = new Dictionary<string, string>
+                Func<string, MoveToErrorsExecutor> moveToErrorsExecutorFactory = localAddress =>
                 {
-                    {FaultsHeaderKeys.FailedQ, localAddress},
-                    {Headers.ProcessingMachine, RuntimeEnvironment.MachineName },
-                    {Headers.ProcessingEndpoint, context.Settings.EndpointName()},
-                    {Headers.HostId, hostInfo.HostId.ToString("N")},
-                    {Headers.HostDisplayName, hostInfo.DisplayName}
+                    var hostInfo = b.Build<HostInformation>();
+                    var staticFaultMetadata = new Dictionary<string, string>
+                    {
+                        {FaultsHeaderKeys.FailedQ, localAddress},
+                        {Headers.ProcessingMachine, RuntimeEnvironment.MachineName},
+                        {Headers.ProcessingEndpoint, context.Settings.EndpointName()},
+                        {Headers.HostId, hostInfo.HostId.ToString("N")},
+                        {Headers.HostDisplayName, hostInfo.DisplayName}
+                    };
+
+                    return new MoveToErrorsExecutor(b.Build<IDispatchMessages>(), errorQueue, staticFaultMetadata);
                 };
 
-                var recoveryActionExecutor = new MoveToErrorsActionExecutor(b.Build<IDispatchMessages>(), errorQueue, staticFaultMetadata);
-
-                var errorBehavior = new MoveFaultsToErrorQueueHandler(
-                    b.Build<CriticalError>(),
-                    failureInfoStorage,
-                    recoveryActionExecutor);
-
-                SecondLevelRetriesHandler slrHandler = null;
-
-                if (IsDelayedRetriesEnabled(context.Settings))
+                var delayedRetriesEnabled = IsDelayedRetriesEnabled(context.Settings);
+                Func<string, DelayedRetryExecutor> delayedRetryExecutorFactory = localAddress =>
```

Comment:
We need to create those executors in pairs if possible for each recoverability executor and get the address of the instance specific queue or the satellite address. Do you have a better proposal? We thought about a better approach but all we could come up with required A LOT more work.